src/main/java/org/servantscode/commons/rest/LoggingFilter.java

```diff
@@ -0,0 +1,104 @@
+package org.servantscode.commons.rest;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.ThreadContext;
+import org.glassfish.jersey.message.internal.ReaderWriter;
+import javax.annotation.Priority;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.Provider;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+@Provider
+@Priority(5000)
+public class LoggingFilter implements ContainerResponseFilter, ContainerRequestFilter {
+    private static final Logger LOG = LogManager.getLogger(LoggingFilter.class);
+
+    @Override
+    public void filter(ContainerRequestContext requestContext,
+                       ContainerResponseContext responseContext) {
+        StringBuilder sb = new StringBuilder("HTTP RESPONSE : ");
+        sb.append("Header: ").append(responseContext.getHeaders());
+        sb.append(" - Entity: ");
+        if(responseContext.getEntity() != null )
+            sb.append(responseContext.getEntity().toString());
```

Comment:
This won't eat the payload will it? What gets logged if the output is a stream?

Reply:
It hasn't removed the payload. I think this explains it.
https://stackoverflow.com/questions/25332103/capture-response-payload-in-jax-rs-filter